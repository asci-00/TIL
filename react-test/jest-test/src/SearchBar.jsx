import React, { useState } from 'react';
import {InputGroup, FormControl, Button, Alert} from 'react-bootstrap'

export default function Component(props) {
  const { onChange, onSubmit } = props
  const [val, setValue] = useState('')

  return (
    <form id='search-bar'>
        <InputGroup className="mb-3">
            <FormControl
                placeholder="Recipient's username"
                aria-label="Recipient's username"
                aria-describedby="basic-addon2"
                value={val}
                onChange={ev => { setValue(ev.target.value); onChange(ev.target.value) }}
            />
            <Button variant="outline-secondary" id="button-addon2" onClick={()=>onSubmit(val)}>Search</Button>
        </InputGroup>
        <Alert variant='dark'>{val ? val : 'Input your text'}</Alert>
    </form>
  )
}