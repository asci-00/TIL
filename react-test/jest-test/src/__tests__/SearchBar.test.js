import { render, screen } from '@testing-library/react';
import SearchBar from '../SearchBar';

describe('Search 시나리오', () => {
  it('DOM 구조 확인', () => {
    const element = render(<SearchBar />);
    screen.debug()
  });
  it('스냅샷 확인', () => {
    const element = render(<SearchBar />);
    expect(element.container).toMatchSnapshot()
  })
  
})

