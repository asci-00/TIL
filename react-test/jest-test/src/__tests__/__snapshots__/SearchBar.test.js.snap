// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search 시나리오 스냅샷 확인 1`] = `
<div>
  <form
    id="search-bar"
  >
    <div
      class="mb-3 input-group"
    >
      <input
        aria-describedby="basic-addon2"
        aria-label="Recipient's username"
        class="form-control"
        placeholder="Recipient's username"
        value=""
      />
      <button
        class="btn btn-outline-secondary"
        id="button-addon2"
        type="button"
      >
        Search
      </button>
    </div>
    <div
      class="fade alert alert-dark show"
      role="alert"
    >
      Input your text
    </div>
  </form>
</div>
`;
